Polygon.js

```
@@ -0,0 +1,185 @@
+	else return null;
+}
+
+function detectWeb(doc, url) {
+	switch (pageType(doc, url)) { 
+		case "Article":
+			return "webpage";
+		case "HubPage":
+			return "multiple";
+	}
+	if (url.indexOf("/search?q=")!=-1){ return "multiple" }
+	else return null;
+}
+
+function scrape(doc, url) {
+	var newItem = new Zotero.Item("webpage");
+	newItem.websiteTitle = "Polygon";
+	newItem.language = "en-US";
+
+	newItem.url = url;
+	newItem.title = ZU.xpathText(doc, '//h2[@class="m-entry__title"]');
+	newItem.date = ZU.xpathText(doc, '//meta[@property="article:published_time"]/@content').split("T")[0];
+
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@name="description"]/@content');
+	newItem.attachments.push({document:doc, title:"Polygon snapshot" ,mimeType:"text/html"});
+
+		// Authors
+	if ( authorMetadata = ZU.xpathText(doc, '//script[contains(text(), "dataLayer =")]') ) {
+		authorMetadata = authorMetadata.match(/"Author"\s?:\s?"([^"]*)"/)[1].split(":");
+		do {
+			newItem.creators.push(ZU.cleanAuthor(authorMetadata[0], "author"));
+			authorMetadata.shift();
+		}
+		while (authorMetadata.length);
+	}
+	
+	newItem.complete();
+}
+
+function doWeb(doc, url) {
+	switch (detectWeb(doc, url)) { 
+		case "webpage":
+			return scrape(doc, url);
+		case "multiple":
+			var items = {};
+	        articles = [];
+	        var titles = ZU.xpath(doc, '//*[contains(concat(" ", normalize-space(@class), " "), "m-block__meta")]/h2/a') //this is needed to catch the class attribute when among others, e.g., "m-block__meta meta" on search pages
+	        for (var i = 0; i < titles.length; i++) {
+	            items[titles[i].href] = titles[i].textContent;
+	        }
+	        Zotero.selectItems(items, function(items) {
+	            if (!items) {
+	                return true;
+	            }
+	            for (var i in items) {
+	                articles.push(i);
+	            }
+	            Zotero.Utilities.processDocuments(articles, scrape);
+	        })
+		default:
```

actually, come to think of it -- I think the desirable behavior for actually be to fail noisily and not silently: From the Zotero perspective, if a user clicks the URL bar icon, it is much better to throw a visible error than to fail silently, because they may not notice the latter. That's different from partial failures to import, in which case the user experience of maybe missing authors is preferable to the whole thing breaking (though those are sometimes judgment calls: e.g. we have Zotero purpusefully fail when items don't have a title).
So yeah, not liking the precendent of the default: return null, even if it'll never fire in this case, sorry.